cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesAdder.java

```diff
@@ -127,22 +309,106 @@ private static PropertyBag createPowerFlowProperties(CgmesModel cgmes, Terminal
         return p;
     }
 
+    // added TopologicalIsland as it was in cgmes : original topology is
+    // preserved.
+    private void addTopologicalIslandsToCgmes() {
+        if (!originalTopologicalIslands.isEmpty()) {
```

Comment:
This test is not necessary, the for-each at line 323 seems sufficient. If it's empty the `byTopologicalIslandId will be empty, and the for-each loop at line 336 won't loop.

Reply:
Done